pkg/ipam/allocator/clusterpool/clusterpool.go

```diff
@@ -61,35 +62,43 @@ type AllocatorOperator struct {
 
 // Init sets up Cilium allocator based on given options
 func (a *AllocatorOperator) Init() error {
-	if len(operatorOption.Config.IPAMOperatorV4CIDR) != 0 {
-		if !option.Config.EnableIPv4 {
-			return errors.New("IPv4CIDR can not be set if IPv4 is not enabled")
+	if option.Config.EnableIPv4 {
+		if len(operatorOption.Config.ClusterPoolIPv4CIDR) == 0 {
+			return fmt.Errorf("%s must be provided when using ClusterPool", operatorOption.ClusterPoolIPv4CIDR)
 		}
-		v4Allocators, err := newCIDRSets(false, operatorOption.Config.IPAMOperatorV4CIDR, operatorOption.Config.NodeCIDRMaskSizeIPv4)
+
+		v4Allocators, err := newCIDRSets(false, operatorOption.Config.ClusterPoolIPv4CIDR, operatorOption.Config.NodeCIDRMaskSizeIPv4)
 		if err != nil {
 			return fmt.Errorf("unable to initialize IPv4 allocator %w", err)
 		}
 		a.v4CIDRSet = v4Allocators
+	} else if len(operatorOption.Config.ClusterPoolIPv4CIDR) != 0 {
+		return errors.New("ClusterPool IPv4 CIDR can not be set if IPv4 is disabled")
```

Comment:
nit: Can we use `operatorOption.ClusterPoolIPv4CIDR` instead ? And s/can/must/g ? And maybe fmt.Errorf() instead of errors.New() for consistency with above lines ?